rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JDataset.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import org.apache.commons.rdf.api.Dataset;
+import org.apache.commons.rdf.api.Quad;
+
+
+/**
+ * Marker interface for RDF4J implementations of Dataset.
+ * 
+ */
+public interface RDF4JDataset extends Dataset, RDF4JGraphLike<Quad> {
```

Comment:
I think it would be possible to expand `ModelGraphImpl` to also support contexts and even a `ModelDataset` - so I'll add that to Jira.  

I think it makes sense for now that it keeps the current behaviour which uses an empty varargs array underneath (aka the "union graph" or "disregarding contexts") - the first goal should be to have "equivalent" integration.